core/domain/collection_domain_test.py

```diff
@@ -643,10 +643,11 @@ def test_yaml_import_and_export(self):
 
         # Should not be able to create a collection from no YAML content.
         with self.assertRaisesRegexp(
-            Exception, 'Please ensure that you are uploading a YAML text file, '
+            utils.InvalidInputException,
+            'Please ensure that you are uploading a YAML text file, '
             'not a zip file. The YAML parser returned the following error: '
-            '\'NoneType\' object has no attribute \'read\''):
-            collection_domain.Collection.from_yaml('collection3', None)
+        ):
+            collection_domain.Collection.from_yaml('collection3', '')
```

Comment:
`None` will fail more noisily. That was done on purpose I think if we call `from_yaml` it should fail noisily since that is not expected argument here.